bigtable/internal/testproxy/proxy.go

```diff
@@ -554,7 +555,52 @@ func (s *goTestProxyServer) RemoveClient(ctx context.Context, req *pb.RemoveClie
 // ReadRow responds to the ReadRow RPC. This method gets all of the column
 // data for a single row in the Table.
 func (s *goTestProxyServer) ReadRow(ctx context.Context, req *pb.ReadRowRequest) (*pb.RowResult, error) {
-	return nil, stat.Error(codes.Unimplemented, "ReadRow not implemented")
+	s.clientsLock.RLock()
```

Comment:
In this case, since you're only ready the `clientIDs` map to get a client, you don't need to hold the lock for the duration of the method.

Instead you can do this:

``` golang
s.clientsLock.RLock()
btc, exists := s.clientIDs[req.ClientId]
s.clientsLock.RUnlock()
```

Reply:
Done.